apps/Unciv/install

```diff
@@ -17,34 +17,42 @@ fi
 # so, we will use Temurin OpenJDK 8 installation cause thats the minimal Java version Unciv needs
 # Temurin very smoothly on Raspberry Pi 3B+ without any problems
 
-status "Adding Adoptium repository:"
+if [[ ! -f /usr/share/keyrings/adoptium-archive-keyring.gpg ]]; then
+  status "Adding Adoptium repository:"
 
-echo "- public key -> keyring"
-rm -f /tmp/adoptium-public-key /tmp/adoptium-archive-keyring.gpg
-wget -O /tmp/adoptium-public-key https://adoptium.jfrog.io/artifactory/api/security/keypair/default-gpg-key/public
-gpg --no-default-keyring --keyring /tmp/adoptium-keyring.gpg --import /tmp/adoptium-public-key
-rm -f /tmp/adoptium-public-key
+  echo "- public key -> keyring"
+  rm -f /tmp/adoptium-public-key /tmp/adoptium-archive-keyring.gpg
+  wget -O /tmp/adoptium-public-key https://adoptium.jfrog.io/artifactory/api/security/keypair/default-gpg-key/public
+  gpg --no-default-keyring --keyring /tmp/adoptium-keyring.gpg --import /tmp/adoptium-public-key
+  rm -f /tmp/adoptium-public-key
 
-echo " - keyring -> GPG key"
-gpg --no-default-keyring --keyring /tmp/adoptium-keyring.gpg --export --output /tmp/adoptium-archive-keyring.gpg
-rm -f /tmp/adoptium-keyring.gpg
+  echo " - keyring -> GPG key"
+  gpg --no-default-keyring --keyring /tmp/adoptium-keyring.gpg --export --output /tmp/adoptium-archive-keyring.gpg
+  rm -f /tmp/adoptium-keyring.gpg
 
-echo " - Moving GPG key to /usr/share/keyrings"
-sudo mv -f /tmp/adoptium-archive-keyring.gpg /usr/share/keyrings
+  echo " - Moving GPG key to /usr/share/keyrings"
+  sudo mv -f /tmp/adoptium-archive-keyring.gpg /usr/share/keyrings
 
-echo " - Creating /etc/apt/sources.list.d/adoptium.list"
-echo "deb [signed-by=/usr/share/keyrings/adoptium-archive-keyring.gpg] https://adoptium.jfrog.io/artifactory/deb $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/adoptium.list >/dev/null
+  echo " - Creating /etc/apt/sources.list.d/adoptium.list"
+  echo "deb [signed-by=/usr/share/keyrings/adoptium-archive-keyring.gpg] https://adoptium.jfrog.io/artifactory/deb $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/adoptium.list >/dev/null
+fi
 
-echo " - Installing temurin-8-jdk"
+status "Installing temurin-8-jdk"
 #try to install temurin-8-jdk; if it fails, remove repository to avoid breaking user's system
 (install_packages temurin-8-jdk)
-if [ $? != 0 ];then
+if [ $? != 0 ]; then
   sudo rm -f /etc/apt/sources.list.d/adoptium.list /usr/share/keyrings/adoptium-archive-keyring.gpg
   error "Failed to install temurin packages. Adoptium repository has been removed."
 fi
 
-### Download Unciv.jar
-wget -O "$HOME/Unciv/Unciv.jar" https://github.com/yairm210/Unciv/releases/download/3.18.10-patch1/Unciv.jar || exit 1
+status "Downloading Unciv.jar"
+echo '- writing into /tmp/Unciv.jar'
+wget -c -O /tmp/Unciv.jar https://github.com/yairm210/Unciv/releases/download/3.18.11/Unciv.jar \
```

Comment:
While adding the `-c` flag may make sense if we were really running the `wget` utility, that's not what's happening here.
Pi-Apps has a _function_ named `wget`. This function automatically switches to use `aria2c` - it's faster, more reliable, and continues downloads by default.
In some cases, `aria2c` cannot completely replace wget, due to how each download utility works. That's why this function will fallback to use `wget` if any unknown flags are passed. In this case, you've added the `-c` flag, and that will cause the `wget` function to fallback to use the `wget` command.

Reply:
You really have an unique naming sense. If you were making function that `get` anyways. then name it as something like `download` so that people can read its description from the documentaion !!! why wrap `aria2c` as `wget` !!
Anyways, I will read `aria2c` docs and edit that.

Reply:
Ok, I read the docs, it seems that the `-c` flag is not necessary here. Although I would personally prefer `wget` here. I am removing it in https://github.com/Botspot/pi-apps/pull/1289/commits/233a21a95ede8e6feadc7ac5ab793d73208b9223.

Anyways, @Botspot, according to the docs `aria2c` uses `--allow-overwrite` flag, which is problematic.  The behavious of `-c` flag in `aria2c` is not like `wget`. If somehow the script throws error at any point in the script, `aria2c` will download all files again for the flag `--allow-overwrite`. Which seems like a bug to me.

Reply:
> Anyways, @Botspot, according to the docs aria2c uses --allow-overwrite flag, which is problematic. The behavious of -c flag in aria2c is not like wget. If somehow the script throws error at any point in the script, aria2c will download all files again for the flag --allow-overwrite. Which seems like a bug to me.


I made a PR to remove the `-c` flag before but I don't know why the `-c` flag is added again...